src/cpu/flexcpu/simple_dataflow_thread.hh

```diff
@@ -0,0 +1,620 @@
+     */
+    bool hasNextPC();
+
+    // END Internal functions
+
+  public:
+
+    // Fullsystem mode constructor
+    SDCPUThread(SimpleDataflowCPU* cpu_, ThreadID tid_, System* system_,
+                     BaseTLB* itb_, BaseTLB* dtb_, TheISA::ISA* isa_,
+                     bool use_kernel_stats_ = true);
+
+    // Non-fullsystem constructor
+    SDCPUThread(SimpleDataflowCPU* cpu_, ThreadID tid_, System* system_,
+                     Process* process_, BaseTLB* itb_, BaseTLB* dtb_,
+                     TheISA::ISA* isa_);
+
+    // May need to define move constructor, due to how SimpleThread is defined,
+    // if we want to hold instances of these in a vector instead of pointers
+    // gotten from new.
+
+    virtual ~SDCPUThread();
+
+    ThreadContext* getThreadContext();
+
+    std::string& name()
+    { return _name; }
+    const std::string& name() const
+    { return _name; }
+
+    void startup();
+
+    // BEGIN ThreadContext functions
+
+    BaseCPU *getCpuPtr() override { return _committedState->getCpuPtr(); }
+
+    int cpuId() const override { return _committedState->cpuId(); }
+
+    uint32_t socketId() const override { return _committedState->socketId(); }
+
+    int threadId() const override { return _committedState->threadId(); }
+
+    void setThreadId(int id) override { _committedState->setThreadId(id); }
+
+    int contextId() const override { return _committedState->contextId(); }
+
+    void setContextId(int id) override { _committedState->setContextId(id); }
+
+    BaseTLB *getITBPtr() override { return _committedState->getITBPtr(); }
+
+    BaseTLB *getDTBPtr() override { return _committedState->getDTBPtr(); }
+
+    CheckerCPU *getCheckerCpuPtr() override
+    { return _committedState->getCheckerCpuPtr(); }
+
+    TheISA::Decoder *getDecoderPtr() override
+    { return &decoder; }
+
+    System *getSystemPtr() override { return _committedState->getSystemPtr(); }
+
+    TheISA::Kernel::Statistics *getKernelStats() override
+    { return _committedState->getKernelStats(); }
+
+    PortProxy &getPhysProxy() override
+    { return _committedState->getPhysProxy(); }
+
+    FSTranslatingPortProxy &getVirtProxy() override
+    { return _committedState->getVirtProxy(); }
+
+    void initMemProxies(ThreadContext *tc) override
+    { _committedState->initMemProxies(tc); }
+
+    SETranslatingPortProxy &getMemProxy() override
+    { return _committedState->getMemProxy(); }
+
+    Process *getProcessPtr() override
+    { return _committedState->getProcessPtr(); }
+
+    void setProcessPtr(Process *p) override
+    { _committedState->setProcessPtr(p); }
+
+    Status status() const override { return _committedState->status(); }
+
+    void setStatus(Status new_status) override
+    { _committedState->setStatus(new_status); }
+
+    /// Set the status to Active.
+    void activate() override;
+
+    /// Set the status to Suspended.
+    void suspend() override { _committedState->suspend(); }
```

Comment:
Probably want to `panic` here. Suspending isn't going to work if there are instructions in the pipeline.